R/checkSummations.R

```diff
@@ -91,15 +101,25 @@ checkSummations <- function(mifFile, outputDirectory = ".", template = "AR6", su
       arrange(tmp, desc(abs(!!sym("reldiff")))), sep = ";",
       file = dataDumpFile, quote = FALSE, row.names = FALSE)
   }
+
   # generate human-readable summary of larger differences
-  .checkSummationsSummary(mifFile, data, tmp, template, summationsFile, summationGroups,
-                   generatePlots, outputDirectory, logFile, logAppend, dataDumpFile, remindVar, plotprefix)
+  .checkSummationsSummary(
+    mifFile, data, tmp, template, summationsFile, summationGroups,
+    generatePlots, outputDirectory, logFile, logAppend, dataDumpFile, remindVar, plotprefix
+  )
 
   return(invisible(tmp))
 }
 
-.checkSummationsSummary <- function(mifFile, data, tmp, template, summationsFile, summationGroups,
-                             generatePlots, outputDirectory, logFile, logAppend, dataDumpFile, remindVar, plotprefix) {
+.checkSummationsSummary <- function(mifFile, data, tmp, template, summationsFile, summationGroups, # nolint: cyclocomp_linter
+                             generatePlots, outputDirectory, logFile, logAppend, dataDumpFile,
+                             remindVar, plotprefix) {
+
+  # does not work for extraction of summations from variables
+  if (summationsFile == "extractVariableGroups") {
```

Comment:
to avoid that this fails with `summationsFile = NULL`:
```suggestion
    if (isTRUE(summationsFile == "extractVariableGroups")) {
```
I don't know whether this is actually possible here, but better safe then sorry ;)

Can you tell me what is not working with the human-readable summary? I generally find that a useful feature, and maybe it can be fixed such that it works?

Reply:
The actual problem is that summationGroups does not exist in this case.

Reply:
Can you try to replace `summationGroups` by `checkVariables` in the call to `.checkSummationsSummary` and replace [this line](https://github.com/pik-piam/piamInterfaces/blob/8e87a5046298df729e46a64436c506701a9e8dda/R/checkSummations.R#L152) by `childs <- checkVariables[p]`?

Reply:
seems to work